server/sonar-server/src/main/java/org/sonar/server/computation/measure/BatchMeasureToMeasure.java

```diff
@@ -109,10 +132,31 @@
       Optional<Measure.Level> qualityGateStatus = Measure.Level.toLevel(batchMeasure.getAlertStatus());
       if (qualityGateStatus.isPresent()) {
         String text = batchMeasure.hasAlertText() ? batchMeasure.getAlertText() : null;
-        measure.setQualityGateStatus(new Measure.QualityGateStatus(qualityGateStatus.get(), text));
+        measure.setQualityGateStatus(new QualityGateStatus(qualityGateStatus.get(), text));
       }
     }
+    if (hasAnyVariation(batchMeasure))  {
+      measure.setVariations(createVariations(batchMeasure));
+    }
     return Optional.of((Measure) measure);
   }
 
+  private static boolean hasAnyVariation(BatchReport.Measure batchMeasure) {
+    return batchMeasure.hasVariationValue1()
+        || batchMeasure.hasVariationValue2()
+        || batchMeasure.hasVariationValue3()
+        || batchMeasure.hasVariationValue4()
+        || batchMeasure.hasVariationValue5();
+  }
+
+  private static MeasureVariations createVariations(BatchReport.Measure batchMeasure) {
+    return new MeasureVariations(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1067)